src/main/java/com/melon/portfoliomanager/repositories/PortfolioItemRepository.java

```diff
@@ -0,0 +1,16 @@
+package com.melon.portfoliomanager.repositories;
+
+import com.melon.portfoliomanager.models.PortfolioItem;
+import com.melon.portfoliomanager.models.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public interface PortfolioItemRepository extends JpaRepository<PortfolioItem, Long> {
+
+    List<PortfolioItem> findByUserAndCompanyName(User user, String companyName);
```

Comment:
```findByUserAndCompanyName``` - can not we search by user_id and company name ? What is ```user``` in the databaes table ? Or this is coming from the the usage of @ManyToOne and related annotations in your domain models ?

Reply:
We are fetching the user to check if such user exists and if such transaction is possible. Since we have already fetched the user, it we can directly query by the object itself and the company name

Reply:
In the DB, we are storing the userID, so when it queries the DB, it's actually searching by the id.